src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -462,16 +469,19 @@ def execute(self):
             classpath_product.remove_for_target(cc.target, [(conf, cc.classes_dir)])
             classpath_product.add_for_target(cc.target, [(conf, cc.jar_file)])
 
+      if self.context.products.is_required_data("export_dep_as_jar_classpath"):
```

Comment:
Could you add a comment here:
```
Because 'runtime classpath' is always a superset of 'export_dep_as_jar_classpath', it is safe to use 'runtime_classpath' as 'export_dep_as_jar_classpath'
```